src/daft-execution/src/ops/limit.rs

```diff
@@ -0,0 +1,53 @@
+use std::sync::Arc;
+
+use common_error::DaftResult;
+use daft_micropartition::MicroPartition;
+use daft_plan::ResourceRequest;
+
+use crate::partition::partition_ref::PartitionMetadata;
+
+use super::PartitionTaskOp;
+
+/// Limit task op.
+#[derive(Debug)]
+pub struct LimitOp {
+    limit: usize,
+    resource_request: ResourceRequest,
+}
+
+impl LimitOp {
+    pub fn new(limit: usize) -> Self {
+        Self {
+            limit,
+            resource_request: ResourceRequest::new_internal(Some(1.0), None, None),
+        }
+    }
+}
+
+impl PartitionTaskOp for LimitOp {
+    type Input = MicroPartition;
+
+    fn execute(&self, inputs: Vec<Arc<Self::Input>>) -> DaftResult<Vec<Arc<MicroPartition>>> {
+        assert_eq!(inputs.len(), 1);
+        let input = inputs.into_iter().next().unwrap();
+        let out = input.head(self.limit)?;
+        Ok(vec![Arc::new(out)])
+    }
+
+    fn resource_request(&self) -> &ResourceRequest {
+        &self.resource_request
+    }
+
+    fn partial_metadata_from_input_metadata(
+        &self,
+        input_meta: &[PartitionMetadata],
+    ) -> PartitionMetadata {
+        assert_eq!(input_meta.len(), 1);
+        let input_meta = &input_meta[0];
```

Comment:
For sure, this is currently unused, the partial metadata machinery is still a TODO for all ops and the surrounding execution model, since no exchange or sink ops have needed it yet.

Reply:
Hmm our current behavior is to set `size_bytes` to `None`, since we don't know the exact new size in bytes until we apply the limit (e.g. you could remove 99% of the rows, but 99% of the size in bytes could be in that remaining 1% of rows if those rows are particularly large).

https://github.com/Eventual-Inc/Daft/blob/021b103cbd3529d1b1acecdc79f7311277c2f0ee/daft/execution/execution_step.py#L552